src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/illegalidentifiername/Example1.java

```diff
@@ -0,0 +1,27 @@
+/*xml
+<module name="Checker">
+  <module name="TreeWalker">
+    <module name="IllegalIdentifierName"/>
+  </module>
+</module>
+*/
+
+//non-compiled with javac: Compilable with Java17
```

Comment:
You can try your approach of adding an `'// ok, because....'` comment. However, I'm not sure if that is going to work.
Please do try once.

Reply:
Done, I added another underscore and put the comment: "// ok, because underscore is part of another word"